### PR TITLE
SONARJAVA-873 Update outdated rules (tags, sqale, description)

### PR DESCRIPTION
**Pull request by benzonico**
Update outdated rules.

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/MethodNamedHashcodeCheck.java [hunk 1]
```diff
@@ -33,8 +33,8 @@
 
 @Rule(
   key = "S1221",
-  name = "Methods should not be named \"hashcode\"",
-  tags = {"pitfall"},
+  name = "Methods should not be named \"hashcode\" or \"equal\"",
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/MethodNamedHashcodeCheck.java]: Check does not look at `equal` at the moment!

File: java-checks/src/main/java/org/sonar/java/checks/DuplicateConditionIfElseIfCheck.java [hunk 2]
```diff
@@ -37,8 +37,8 @@
 
 @Rule(
   key = DuplicateConditionIfElseIfCheck.RULE_KEY,
-  name = "Conditions in related \"if/else if\" statements should not be duplicated",
-  tags = {"bug", "unused"},
+  name = "Conditions in related \"if/else if\" statements and \"cases\" in a \"(switch|Select Case)\" should not have the same condition",
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/DuplicateConditionIfElseIfCheck.java]: Check currently does not verify switch statements!

File: java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S1125.html [hunk 3]
```diff
@@ -1,21 +1,30 @@
...
+<p>Remove literal boolean values from conditional expressions to improve readability. Anything that can be tested for equality with a boolean value must itself be a boolean value, and boolean values can be tested atomicly.</p>
 <h2>Noncompliant Code Example</h2>
 
 <pre>
-if (booleanVariable == true) { /* ... */ }
-if (booleanVariable != true) { /* ... */ }
-if (booleanVariable || false) { /* ... */ }
+if (booleanVariable == true) { /* ... */ }      
+if (booleanVariable != true) { /* ... */ }   
+if (booleanVariable || false) { /* ... */ }
```
Comment (m-g-sonar) [inline java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S1125.html]: Why the trailing spaces?

File: java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S1125.html [hunk 4]
```diff
@@ -1,21 +1,30 @@
...
+</pre>
 <h2>Compliant Solution</h2>
 
 <pre>
-if (booleanVariable) { /* ... */ }
-if (!booleanVariable) { /* ... */ }
-if (booleanVariable) { /* ... */ }
+if (booleanVariable) { /* ... */ }              
+if (!booleanVariable) { /* ... */ }    
+if (booleanVariable) { /* ... */ }
```
Comment (m-g-sonar) [inline java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S1125.html]: Why the trailing spaces?

File: java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S1150.html [hunk 5]
```diff
@@ -1,23 +1,17 @@
...
-New implementations should consider using Iterator in preference to Enumeration.
+<p>NOTE: The functionality of this Enumeration interface is duplicated by the Iterator interface. In addition, Iterator adds an optional remove operation, and has shorter method names. New implementations should consider using Iterator in preference to Enumeration.</p>
 </blockquote>
-
 <p>The following code:</p>
-
 <pre>
 public class MyClass implements Enumeration {  // Non-Compliant
-  /* ... */
+  /* … */
```
Comment (m-g-sonar) [inline java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S1150.html]: not sure about using the 3 points character.

File: java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S1150.html [hunk 6]
```diff
@@ -1,23 +1,17 @@
...
 }
 </pre>
-
 <p>should be refactored into:</p>
-
 <pre>
 public class MyClass implements Iterator {     // Compliant
-  /* ... */
+  /* … */
```
Comment (m-g-sonar) [inline java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S1150.html]: not sure about using the 3 points character.

File: java-checks/src/main/java/org/sonar/java/checks/MethodNamedHashcodeCheck.java (hunk 1 above)
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/MethodNamedHashcodeCheck.java]: You're right, we have a dedicated check for that (S1201). RSPEC updated.

File: java-checks/src/main/java/org/sonar/java/checks/DuplicateConditionIfElseIfCheck.java (hunk 2 above)
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/DuplicateConditionIfElseIfCheck.java]: RSPEC updated for Java, a switch statement with twice the same case won't compile.

File: java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S1125.html (hunk 3 above)
Comment (benzonico) [inline java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S1125.html]: Generated.

**Comment by m-g-sonar** [comment]
LGTM.

**Pull request closed without merging**